packages/client/src/sync/fetcher/accountfetcher.ts

```diff
@@ -218,6 +154,129 @@ export class AccountFetcher extends Fetcher<JobTask, AccountData[], AccountData>
     )
   }
 
+  async fetch(): Promise<boolean> {
+    this.fetcherDoneFlags.syncing = true
+
+    try {
+      this.fetcherDoneFlags.accountFetcher.started = true
+      const accountFetch = !this.fetcherDoneFlags.accountFetcher.done ? super.fetch() : null
+      // wait for all accounts to fetch else storage and code fetcher's doesn't get us full data
+      this.config.superMsg(`Snapsync: running accountFetch=${accountFetch !== null}`)
+      await accountFetch
+      this.fetcherDoneFlags.accountFetcher.started = false
+      if (this.fetcherDoneFlags.accountFetcher.done !== true) {
+        // @scorbajio need to see the reason for this here
+        throw Error('accountFetcher finished without completing the sync')
+      }
+
+      const storageFetch = !this.fetcherDoneFlags.storageFetcherDone
+        ? this.storageFetcher.fetch().then(
+            // we should not be doing this, fetcher itself should mark completion
```

Comment:
An alternative to using callbacks for calling `snapFetchersCompleted` is making `snapFetchersCompleted` available in the `storagefetcher` by either exporting it or passing it into it as a parameter. Then, it can call it in store when it receives the termination signal and is setting `destroyWhenDone` to `true`:
```
if (JSON.stringify(result[0]) === JSON.stringify(Object.create(null))) {
        this.debug(
          'Empty result detected - Associated range requested was empty with no elements remaining to the right'
        )
        this.destroyWhenDone = true
        return
}
```
However, the code and trienode fetchers don't have a clear terminating condition. That's why they have the `setDestroyWhenDone` function implemented so that when the account fetcher is done and all other necessary nodes and codes have been queued for fetching, those two fetchers will be destroyed as soon as they complete their last job.